pyanaconda/modules/storage/partitioning/custom/custom_partitioning.py

```diff
@@ -1097,6 +1097,10 @@ def _execute_logvol_data(self, storage, data, logvol_data):
 
             if logvol_data.cache_size and logvol_data.cache_pvs:
                 pv_devices = [lookup_alias(devicetree, pv) for pv in logvol_data.cache_pvs]
+                # for encrypted pvs the names are associated with the underlying partition
+                # but we need the LUKS device on top of it
+                pv_devices = [pv.children[0] if pv.format.type == "luks" else pv
```

Comment:
Do you think you could rewrite the code that creates `pv_devices` and use loops instead of generators?